String not null terminated

nishith-vihar opened the pull request:
CID: 1214629,1274235,1437648
The buffer has been null terminated thus resolving the issue

Change-Id: Ieb1d067d8dd860c55a8091dd6fbba1bcbb4dc19f
Updates: #1060
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>

nishith-vihar committed:
String not null terminated
CID: 1214629,1274235,1437648
The buffer has been null terminated thus resolving the issue

Change-Id: Ieb1d067d8dd860c55a8091dd6fbba1bcbb4dc19f
Updates: #1060
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

nik-redhat commented:
/run regression

schaffung commented:
/run regression

aspandey commented on xlators/mgmt/glusterd/src/glusterd-geo-rep.c:
```diff
@@ -4422,6 +4422,7 @@ glusterd_gsync_read_frm_status(char *path, char *buf, size_t blen)
         return -1;
     }
     ret = sys_read(status_fd, buf, blen - 1);
+    buf[ret] = '\0';
```
In case of failure. I think sys_read will return -1 or negative value which will make this line wrong and cause corruption.

mohit84 replied:
We should set '\0' in case of success only.

nishith-vihar replied:
Oh yeah. I think the null should be added in the 'if' condition which is which is followed after calling sys_read

nishith-vihar committed:
String not Null Terminated
CID: 1214629,1274235,1437648
The buffer has been null terminated thus resolving the issue

Updates: #1060
Change-Id: Ib9c566d2633df5d056ed3554f2c3c29364a28993
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>

nik-redhat commented:
/run regression

schaffung commented:
/run regression
